### DISCUSSION TITLE
Diffusion model

### DISCUSSION THREAD

**Discussion by azibitsker** [mutationpp/Mutationpp#263] [General]
Hello community, 
I had a question regarding the correct use of `ChemNonEqTTv `state model to compute mixture averaged diffusion coefficients.
After creating and initializing `mix_NEQ `object with the `Ramshaw `diffusion model, I do the following:
```
mix_NEQ->setState(spRho.data(), TTv.data(), 1);
mix_NEQ->averageDiffusionCoeffs(Ds.data());
```
where spRho is the species densities vector, TTv is the vector containing T, Tv and Ds is the vector that I fill with diffusion coefficients.
I'm asking this question because I get quite different diffusion coefficients with the Ramshaw model, compared to the Gupta-Yos model (that I use from a different code). 
If the implementation is correct, could you also comment on why I would see the sensible difference in Ds between Ramshaw and Gupta-Yos models? Would the model 'Exact' in M++ library be somewhat similar to the Gupta-Yos model?

**Comment by mgoodson-cvd** [comment]
Hey Alex, I haven't looked into the diffusion matrix code too much to be able to compare to Gupta-Yos, but you should be able to switch from `Ramshaw` (the default) to the `Exact` diffusion matrix using `mix_NEQ->setDiffusionMatrixAlgo("Exact")` and compare results for your case.

**Reply by azibitsker** [reply]
Hey Matthew, 
Right, I don't know why I didn't try it before asking. Running with the `Exact` model shows similar trends as with the `Ramshaw` model. Both models predict higher net diffusive flux, compared to the Gupta-Yos model in my air-carbon ablation test case. Interestingly, the Gupta-Yos model I use, has collision data from Magin2002 paper for CO,CO2,O2, O,C pairs, while M++ uses data from Wright2007.

**Reply by mgoodson-cvd** [reply]
I think you're on the right track in comparing the collision data. I have compared M++ diffusion coefficients to other NASA codes in the past and only observed minor differences. I would try your Gupta-Yos model with the M++ collision data and see what you get, or vice-versa.
